src/ServiceFabricPersistence/Outbox/OutboxStorage.cs

```diff
@@ -111,7 +122,54 @@ public Task<OutboxTransaction> BeginTransaction(ContextBag context)
                     }
 
                     currentIndex++;
-                    cleanConditionalValue = await Cleanup.TryPeekAsync(tx, LockMode.Default, defaultOperationTimeout, cancellationToken).ConfigureAwait(false);
+                    cleanConditionalValue = await CleanupOld.TryPeekAsync(tx, LockMode.Default, defaultOperationTimeout, cancellationToken).ConfigureAwait(false);
+                }
+
+                if (somethingToCommit)
```

Comment:
To refresh my memory (what's that?!) - we've doing conditional transaction commit to avoid exception if there was nothing to commit or for optimization?

Reply:
This is a good question. I think, this was an optimization of the call @SeanFeldman

Reply:
With SF stuff being open-source these days, should we check if this is even needed?
If we can remove unnecessary "dirty tracking" and just commit, relying on the underlying implementaiton to check if anything needs to be committed or not (assuming it does) would make this code simpler.

Reply:
https://stackoverflow.com/questions/50300697/downsides-of-commitasync-w-o-any-changes-to-collection

Reply:
I'm not sure if I'm ready to dive into COM+ IServiceFabricV6 implementation. AFAIK all the managed interfaces are translated to the COM+ stuff so it won't be an easy trip. Still, I'm planning to do this dive in a few weeks so will happily share my findings.

Reply:
Wait, you don't do that as a practice before breakfast 😱 
Haha, no worries. I'm just curiouse.

Reply:
Matt Snider has confirmed that committing a "null" transaction has not performance impact. Any objections if we remove unnecessary `if (somethingToCommit)`?

Reply:
Not from my side. Less code and branching in the code is always a plus